engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -93,550 +114,397 @@
+        this.minY = minY;
+        this.minZ = minZ;
 
-    public BlockRegion copy() {
-        return new BlockRegion(this);
+        this.maxX = maxX;
+        this.maxY = maxY;
+        this.maxZ = maxZ;
+        return this;
     }
 
-    // -- min --------------------------------------------------------------------------------------------------------//
-
     /**
-     * The x-coordinate of the minimum corner
+     * Reset both the minimum and maximum corner of this region.
+     *
+     * @return this region (after modification)
+     * @throws IllegalArgumentException if the given coordinates for min and max are {@link #isValid()
+     *         invalid}.
      */
-    public int minX() {
-        return this.minX;
+    public BlockRegion set(Vector3ic min, Vector3ic max) {
+        return this.set(min.x(), min.y(), min.z(), max.x(), max.y(), max.z());
     }
 
     /**
-     * the minimum coordinate of the first block x
+     * Reset this region to have the same minimum and maximum corner as the {@code source} region.
      *
-     * @return the minimum coordinate x
-     * @deprecated use {@link #minX()}
+     * @return this region (after modification)
+     * @throws IllegalArgumentException if the given coordinates for min and max are {@link #isValid()
+     *         invalid}.
      */
-    @Deprecated
-    public int getMinX() {
-        return this.minX;
+    public BlockRegion set(BlockRegionc source) {
+        return this.set(source.minX(), source.minY(), source.minZ(), source.maxX(), source.maxY(), source.maxZ());
     }
 
-    /**
-     * set the minimum coordinate of the first block x
-     *
-     * @return the minX
-     */
-    public BlockRegion minX(int x) {
-        Preconditions.checkArgument(x <= this.maxX || this.maxX == Integer.MIN_VALUE);
-        this.minX = x;
-        return this;
+    // -- min --------------------------------------------------------------------------------------------------------//
+
+    @Override
+    public int minX() {
+        return this.minX;
     }
 
-    /**
-     * The y-coordinate of the minimum corner
-     */
+    @Override
     public int minY() {
         return this.minY;
     }
 
-    /**
-     * the minimum coordinate of the first block y
-     *
-     * @return the minimum coordinate y
-     * @deprecated use {@link #minY()}
-     */
-    @Deprecated
-    public int getMinY() {
-        return this.minY;
+    @Override
+    public int minZ() {
+        return this.minZ;
     }
 
-    /**
-     * set the minimum coordinate of the first block y
-     *
-     * @return the minY
-     */
-    public BlockRegion minY(int y) {
-        Preconditions.checkArgument(y <= this.maxY || this.maxY == Integer.MIN_VALUE);
-        this.minY = y;
-        return this;
+    @Override
+    public BlockRegion minX(int x, BlockRegion dest) {
+        Preconditions.checkArgument(x <= this.maxX || this.maxX == Integer.MIN_VALUE);
+        dest.minX = x;
+        return dest;
```

Comment:
This implementation is wrong!
```suggestion
    public BlockRegion minX(int x, BlockRegion dest) {
        return dest.set(x, minY, minZ, maxX, maxY, maxZ);
```

Reply:
Alternatively, we could only offer these methods to modify individual components on the mutable class.

Reply:
Guess that's a question for @pollend

Reply:
Why do you need to have a dest for a mutation like this. Just modify the direct object. Simpler to set it to another object and then set min and max.

Reply:
Hm, can we clarify in which situations we want to have the variants with explicit destinations and in which we don't want/need them? I don't see a clear pattern here :see_no_evil: